lib/tapjoy/autoscaling_bootstrap.rb

```diff
@@ -168,6 +171,7 @@ def configure_environment(filename, env=nil)
 
         new_config = defaults_hash.merge!(env_hash).merge(facet_hash)
         new_config[:config_dir] = config_dir
+        new_config[:instance_ids] = opts[:instance_ids] if opts.key?(:instance_ids)
```

Comment:
In this contxt, you almost don't even care what is in opts[:instance_ids].

If opts[:instance_ids] is nil, new_config[:instance_ids] will still end up nil. So you already must be handling that case downstream somewhere.

Since it doesn't matter if either of them are nil, your check for opts.key? or opts[:instance_id] to conditionally set new_config doesn't matter.

TL;DR;

```ruby
new_config[:instance_ids] = opts[:instance_ids]
```

Same great taste, less code to maintain.